postgres/datadog_checks/postgres/statements.py

```diff
@@ -406,25 +456,69 @@ def _emit_pg_stat_statements_metrics(self):
         except psycopg2.Error as e:
             self._log.warning("Failed to query for pg_stat_statements count: %s", e)
 
+    # _apply_deltas expects normalized rows before any merging of duplicates.
+    # It takes the incremental pg_stat_statements rows and constructs the full set of rows
+    # by adding the existing values in the baseline_metrics cache. This is equivalent to
+    # fetching the full set of rows from pg_stat_statements, but we avoid paying the price of
+    # actually querying the rows.
+    def _apply_deltas(self, rows):
+        # Apply called queries to baseline_metrics
+        for row in rows:
+            query_signature = row['query_signature']
+            queryid = row['queryid']
+            baseline_row = copy.copy(row)
+            if query_signature not in self._baseline_metrics:
+                self._baseline_metrics[query_signature] = {queryid: baseline_row}
+            else:
+                self._baseline_metrics[query_signature][queryid] = baseline_row
+
+        # Apply query text, so it doesn't have to be cached.
+        query_text = {row['query_signature']: row['query'] for row in rows}
```

Comment:
this doesn't seem to be necessary because `baseline_row = copy.copy(row)` will have query text cached.

Reply:
I accidentally removed a line where the query text is deleted from this row after copying, that's why this is here. I restored it.

The idea is to avoid caching the query text, and instead get it from only the called queries on each check run. So the rows that get constructed and pass to `_compute_derivative_rows`  will only have the query text if they were called and thus will have metrics sent for them. Queries that weren't called won't get metrics sent, so the query text isn't needed there.